include/linux/double-fetch-detection.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __LINUX_DOUBLE_FETCH_DETECTION_H__
+#define __LINUX_DOUBLE_FETCH_DETECTION_H__
+#ifdef CONFIG_DF_DETECTION
+
+struct address{
+    void *startaddress;
+    unsigned long len;
+};
+
+
+void addAddress(void* addr,unsigned long len);
+void startSystemCall(void);
+void endSysCall(void);
```

Comment:
For paired names use consistent naming, e.g. `endSystemCall` instead of `endSysCall` (or, in our case, `end_system_call`)